quil-rs/src/program/analysis/control_flow_graph.rs

```diff
@@ -269,6 +269,37 @@ pub struct BasicBlockOwned {
     terminator: BasicBlockTerminatorOwned,
 }
 
+impl BasicBlockOwned {
+    pub fn new(
+        label: Option<Target>,
+        instructions: Vec<Instruction>,
+        instruction_index_offset: usize,
+        terminator: BasicBlockTerminatorOwned,
+    ) -> Self {
+        BasicBlockOwned {
+            label,
+            instructions,
+            instruction_index_offset,
+            terminator,
+        }
+    }
+    pub fn label(&self) -> Option<&Target> {
+        self.label.as_ref()
+    }
+
+    pub fn instructions(&self) -> &[Instruction] {
+        &self.instructions
+    }
+
+    pub fn instruction_index_offset(&self) -> usize {
+        self.instruction_index_offset
+    }
+
+    pub fn terminator(&self) -> &BasicBlockTerminatorOwned {
+        &self.terminator
+    }
+}
```

Comment:
Are these needed? the `*Owned` structs here are explicitly just as a holding pen for the inner data because the python wrappers can't just hold a ref. AFAIK they should never be directly constructed or queried but only be converted to/from `BasicBlock`

Reply:
Nah, I thought I had those changes discarded. Removed.